### PR TITLE
Je/refactor find usages

### PR DESCRIPTION
**Pull request by emord**
https://github.com/dimagi/Vellum/commit/3d6811b352a138d4b51d18170b1c181b445fafe6#commitcomment-19687675

classic @millerdev using refactoring as a ruse to make it easier to test :P 

@esoergel

### CODE REVIEW

File: src/logic.js [hunk 1]
```diff
@@ -501,7 +501,46 @@ define([
...
+            _.each(refs, function (value, key) {
+                var usedMug = form.getMugByUFID(key),
+                    mugReferences = {};
+                if (ufidOrPath && ufidOrPath === usedMug.hashtagPath) {
+                    return;
+                }
+                _.each(value, function (value, key) {
+                    _.each(value, function (value) {
+                        var usedInMug = form.getMugByUFID(key),
+                            readablePropName = usedInMug.spec[value.property].lstring;
```
Comment (emord) [inline src/logic.js]: Arguably should return `value.property` and the modal code should change it to a readable name, but I think this is good enough

**Comment by millerdev** [comment]
what else do you use refactoring for?

File: src/logic.js [hunk 2]
```diff
@@ -501,7 +501,42 @@ define([
...
+                refs = _this.reverse;
+
+            _.each(refs, function (value, key) {
+                var usedMug = form.getMugByUFID(key),
+                    mugReferences = {};
+                if (path && path !== usedMug.hashtagPath) {
+                    return;
+                }
+                _.each(value, function (value, key) {
+                    _.each(value, function (value) {
```
Comment (millerdev) [inline src/logic.js]: using `value` ~~twice~~ thrice in nested scopes like this makes it hard to follow. More descriptive names would be nice. Also, same for `key`.

File: tests/logic.js [hunk 3]
```diff
@@ -204,6 +204,33 @@ define([
...
+            before(function (done) {
+                form = util.loadXML(TEST_XML_1);
+                done();
+            });
+
+            it("should return dictionary of usages", function () {
+                var expectedOutput = {
+                        "#form/question1": {"#form/question2": "Display Condition"}
+                    };
+                assert.deepEqual(form.findUsages(), expectedOutput);
```
Comment (millerdev) [inline tests/logic.js]: Maybe load a form with more usages so the expected output of this test is not the same as the one that filters on a path?

File: src/logic.js (hunk 2 above)
Comment (emord) [inline src/logic.js]: slightly better naming although I think it will still take me a bit to figure out what's going on when I come back to it https://github.com/dimagi/Vellum/pull/794/commits/e476b06bea4390dd0bbbf13b901ef20c6aec3108

File: tests/logic.js (hunk 3 above)
Comment (emord) [inline tests/logic.js]: https://github.com/dimagi/Vellum/pull/794/commits/bf04c1c0cdf20408ec7c69c508401c1da7908a8d
